python/dgl/nn/pytorch/network_emb.py

```diff
@@ -0,0 +1,220 @@
+                    for word in line:
+                        self.token_count += 1
+                        word_frequency[word] = word_frequency.get(word, 0) + 1
+
+        # Filter out the number of words less than the minimum frequency
+        t = 0.0001
+        for word, freq in word_frequency.items():
+            if freq > self.min_count:
+                word_frequency_filtered[word] = freq
+                f = np.array(freq) / self.token_count
+                self.discards[word] = np.sqrt(t / f) + (t / f)
+
+        # get the number of all nodes
+        self.word_count = self.g.number_of_nodes()
+        self.word_frequency = word_frequency_filtered
+
+        print("Total embeddings: " + str(self.word_count))
+        print("real embeddings：" + str(len(self.word_frequency)))
+
+        self.u_embeddings = nn.Embedding(self.word_count, self.emb_dimension, sparse=self.sparse)
+        self.v_embeddings = nn.Embedding(self.word_count, self.emb_dimension, sparse=self.sparse)
+
+    def _initTableNegatives(self):
+        # get a table for negative sampling, if word with index 2 appears twice, then 2
+        # will be listed in the table twice.
+        pow_frequency = np.array(list(self.word_frequency.values())) ** 0.75
+        words_pow = sum(pow_frequency)
+        ratio = pow_frequency / words_pow
+        count = np.round(ratio * self.NEGATIVE_TABLE_SIZE)
+        word_list = list(self.word_frequency.keys())
+        for wid, c in enumerate(count):
+            self.negatives += [word_list[wid]] * int(c)
+        self.negatives = np.array(self.negatives)
+        np.random.shuffle(self.negatives)
+        self.sampling_prob = ratio
+
+    def initParameters(self):
+        """
+        Initialize the embedding parameters
+        """
+        initrange = 1.0 / self.emb_dimension
+        init.uniform_(self.u_embeddings.weight.data, -initrange, initrange)
+        init.constant_(self.v_embeddings.weight.data, 0)
+
+    def _getNegatives(self, size):
+        response = self.negatives[self.negpos:self.negpos + size]
+        self.negpos = (self.negpos + size) % len(self.negatives)
+        if len(response) != size:
+            return np.concatenate((response, self.negatives[0:self.negpos]))
+        return response
+
+    def _generate_sample(self, batches):
+        pair_catch = []
+        for batch in batches:
+            if len(batch) < 0:
+                continue
+            word_ids = [w for w in batch if w in self.word_frequency
+                        and np.random.rand() < self.discards[w]]
+
+            for i, u in enumerate(word_ids):
+                for j, v in enumerate(
+                        word_ids[max(i - self.window_size, 0):i + self.window_size]):
+                    assert u < self.word_count
+                    assert v < self.word_count
+                    if i == j:
+                        continue
+                    pair_catch.append((u, v, self._getNegatives(self.negative_samples_size)))
```

Comment:
This is the only place where `_getNegatives` is used and the size is always `self.negative_samples_size`. Then why don't we drop the size argument of `_getNegatives` and let it directly access `self.negative_samples_size`?